distributed/client.py

```diff
@@ -3587,12 +3578,14 @@ def persist(
             fifo_timeout=fifo_timeout,
             actors=actors,
         )
-
-        postpersists = [c.__dask_postpersist__() for c in collections]
-        result = [
-            func({k: futures[k] for k in flatten(c.__dask_keys__())}, *args)
-            for (func, args), c in zip(postpersists, collections)
-        ]
+        if newstyle_collections(collections):
+            result = [var.postpersist(futures) for var in collections]
+        else:
+            postpersists = [c.__dask_postpersist__() for c in collections]
+            result = [
+                func({k: futures[k] for k in flatten(c.__dask_keys__())}, *args)
+                for (func, args), c in zip(postpersists, collections)
+            ]
```

Comment:
We could keep the old `__dask_postpersist__` protocol. I think reversing the control for this API is not as powerful as for postcompute but I would do it like this for symmetry reasons. The only caveat here is that the new collection API technically doesn't know anything about keys so we have to pass all futures to this API and trust the collection to do the right thing